executor/partition_table_test.go

```diff
@@ -683,6 +683,79 @@ func (s *partitionTableSuite) TestDynamicPruningUnderIndexJoin(c *C) {
 		tk.MustQuery(`select /*+ INL_JOIN(touter, tnormal) */ tnormal.* from touter join tnormal use index(idx_b) on touter.b = tnormal.b`).Sort().Rows())
 }
 
+func (s *partitionTableSuite) TestBatchGetforRangeandListPartitionTable(c *C) {
+	if israce.RaceEnabled {
+		c.Skip("exhaustive types test, skip race test")
+	}
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("create database test_pointget")
+	tk.MustExec("use test_pointget")
+	tk.MustExec("set @@tidb_partition_prune_mode = 'dynamic'")
+	tk.MustExec("set @@session.tidb_enable_list_partition = ON")
+
+	// list partition table
+	tk.MustExec(`create table tlist(a int, b int, unique index idx_a(a), index idx_b(b)) partition by list(a)( 
+		partition p0 values in (1, 2, 3, 4),
+			partition p1 values in (5, 6, 7, 8),
+			partition p2 values in (9, 10, 11, 12));`)
+
+	// range partition table
+	tk.MustExec(`create table trange(a int, unique key(a)) partition by range(a) (
+		partition p0 values less than (30),
+		partition p1 values less than (60),
+		partition p2 values less than (90),
+		partition p3 values less than (120));`)
```

Comment:
Please add more cases with unsigned/varchar/datetime/decimal types~

Reply:
It seems that range and list partition can only support `INT` typed partition key. How about we only add tests about unsigned?

Reply:
unsigned type case is updated in the new commit.

Reply:
`partition by range columns(...)` can support other types, but `partition by range columns(...)` doesn't support `BatchGet`.

Reply:
Oh, my fault.